test/starlark_tests/visionos_application_tests.bzl

```diff
@@ -47,6 +90,341 @@ def visionos_application_test_suite(name):
+            name,
+        ],
+        target_compatible_with = select({
+            "//:supports_visionos_setting": [],
+            "//conditions:default": ["@platforms//:incompatible"],
+        }),
+    )
+
+    archive_contents_test(
+        name = "{}_contains_asan_dylib_device_test".format(name),
+        build_type = "device",
+        contains = [
+            "$BUNDLE_ROOT/Frameworks/libclang_rt.asan_xros_dynamic.dylib",
+        ],
+        sanitizer = "asan",
+        target_under_test = "//test/starlark_tests/targets_under_test/visionos:app_minimal",
+        tags = [
+            name,
+        ],
+        target_compatible_with = select({
+            "//:supports_visionos_setting": [],
+            "//conditions:default": ["@platforms//:incompatible"],
+        }),
+    )
+
+    archive_contents_test(
+        name = "{}_contains_tsan_dylib_device_test".format(name),
+        build_type = "simulator",
+        contains = [
+            "$BUNDLE_ROOT/Frameworks/libclang_rt.tsan_xrossim_dynamic.dylib",
+        ],
+        sanitizer = "tsan",
+        target_under_test = "//test/starlark_tests/targets_under_test/visionos:app_minimal",
+        tags = [
+            name,
+        ],
+        target_compatible_with = select({
+            "//:supports_visionos_setting": [],
+            "//conditions:default": ["@platforms//:incompatible"],
+        }),
+    )
+
+    archive_contents_test(
+        name = "{}_contains_ubsan_dylib_simulator_test".format(name),
+        build_type = "simulator",
+        contains = [
+            "$BUNDLE_ROOT/Frameworks/libclang_rt.ubsan_xrossim_dynamic.dylib",
+        ],
+        sanitizer = "ubsan",
+        target_under_test = "//test/starlark_tests/targets_under_test/visionos:app_minimal",
+        tags = [
+            name,
+        ],
+        target_compatible_with = select({
+            "//:supports_visionos_setting": [],
+            "//conditions:default": ["@platforms//:incompatible"],
+        }),
+    )
+
+    archive_contents_test(
+        name = "{}_contains_ubsan_dylib_device_test".format(name),
+        build_type = "device",
+        contains = [
+            "$BUNDLE_ROOT/Frameworks/libclang_rt.ubsan_xros_dynamic.dylib",
+        ],
+        sanitizer = "ubsan",
+        target_under_test = "//test/starlark_tests/targets_under_test/visionos:app_minimal",
+        tags = [
+            name,
+        ],
+        target_compatible_with = select({
+            "//:supports_visionos_setting": [],
+            "//conditions:default": ["@platforms//:incompatible"],
+        }),
+    )
+
+    apple_codesigning_dossier_info_provider_test(
+        name = "{}_codesigning_dossier_info_provider_test".format(name),
+        expected_dossier = "app_dossier.zip",
+        target_under_test = "//test/starlark_tests/targets_under_test/visionos:app",
+        tags = [
+            name,
+        ],
+        target_compatible_with = select({
+            "//:supports_visionos_setting": [],
```

Comment:
All these tests can't run on our CI at the moment. As soon as Xcode 15.1 is out of beta, we should ask to install it on the CI machines to allow running them.